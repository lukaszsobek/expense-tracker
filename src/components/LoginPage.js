import React from "react"
import { connect } from "react-redux"

import { startLogin } from "../actions"

export const LoginPage = ({ startLogin }) => {
    return (
        <div className="login__imageBg">
            <div className="login__contentBox">
                <h1 className="login__title">Expense Tracker</h1>
                <p className="login__slogan">Track your expenses with this app</p>
                <button
                    className="login__button"
                    onClick={startLogin}
                >
                    Log in with Google
                </button>
            </div>
        </div>
    )
}

const mapDispatchToProps = dispatch => ({
    startLogin: () => dispatch(startLogin)
})

export default connect(null, mapDispatchToProps)(LoginPage)