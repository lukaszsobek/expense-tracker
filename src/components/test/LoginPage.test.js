import React from "react"
import { shallow } from "enzyme"

import { LoginPage } from "../LoginPage"

describe("LoginPage",() => {

    it("renders",() => {
        const wrapper = shallow(<LoginPage />)
        expect(wrapper).toMatchSnapshot()
    })

    it("calls correct function on login",() => {
        const startLoginSpy = jest.fn()
        const wrapper = shallow(<LoginPage startLogin={startLoginSpy}/>)
        wrapper.find("button").simulate("click")
        expect(startLoginSpy).toBeCalled()
    })

})
