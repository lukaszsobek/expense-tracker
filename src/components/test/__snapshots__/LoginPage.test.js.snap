// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginPage renders 1`] = `
<div
  className="login__imageBg"
>
  <div
    className="login__contentBox"
  >
    <h1
      className="login__title"
    >
      Expense Tracker
    </h1>
    <p
      className="login__slogan"
    >
      Track your expenses with this app
    </p>
    <button
      className="login__button"
    >
      Log in with Google
    </button>
  </div>
</div>
`;
